src/sentry/static/sentry/app/utils/withTeamsForUser.tsx

```diff
@@ -39,20 +40,34 @@ const withTeamsForUser = <P extends InjectedTeamsProps>(
     }
 
     async fetchTeams() {
+      // check if we can use organization teams/projects instead of fetching data
+      const {projects} = this.props.organization;
+      let {teams} = this.props.organization;
+      if (projects && teams) {
+        const {isSuperuser} = ConfigStore.get('user');
+        const {projectsByTeam} = getProjectsByTeams(teams, projects, isSuperuser);
+        teams.forEach(team => {
+          team.projects = projectsByTeam[team.slug] || [];
```

Comment:
@davidenwang This is going to mutate your props which could produce unintended side effects. We should make a copy (e.g. `Object.assign`) instead.